highway/crud_object.go

```diff
@@ -2,22 +2,158 @@ package highway
 
 import (
 	context "context"
+	"errors"
 
+	otv1 "github.com/sonr-io/blockchain/x/object/types"
 	ot "go.buf.build/grpc/go/sonr-io/blockchain/object"
 )
 
 // CreateObject creates a new object.
 func (s *HighwayServer) CreateObject(ctx context.Context, req *ot.MsgCreateObject) (*ot.MsgCreateObjectResponse, error) {
+	// Verify that object fields are not nil
+	if req.GetInitialFields() == nil {
+		return nil, errors.New("object to register must have fields")
+	}
 
-	return nil, ErrMethodUnimplemented
+	// Translate ot fields to otv1
+	fields := s.bufToBlockchain(req.GetInitialFields())
+
+	// Build Transaction
+	tx := &otv1.MsgCreateObject{
+		Creator:       req.GetCreator(),
+		Label:         req.GetLabel(),
+		Description:   req.GetDescription(),
+		InitialFields: fields,
+	}
+
+	// Broadcast the message
+	res, err := s.cosmos.BroadcastCreateObject(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	// decode otv1 fields to ot
+	outputFields := s.decodeObjectDocFields(res.WhatIs.ObjectDoc.Fields)
+
+	return &ot.MsgCreateObjectResponse{
+		Code:    res.GetCode(),
+		Message: res.GetMessage(),
+		WhatIs: &ot.WhatIs{
+			Did: res.WhatIs.GetDid(),
+			ObjectDoc: &ot.ObjectDoc{
+				Label:       res.WhatIs.ObjectDoc.GetLabel(),
+				Description: res.WhatIs.ObjectDoc.GetDescription(),
+				Did:         res.WhatIs.ObjectDoc.GetDid(),
+				BucketDid:   res.WhatIs.ObjectDoc.GetBucketDid(),
+				Fields:      outputFields,
+			},
+			Creator:   res.WhatIs.GetCreator(),
+			Timestamp: res.WhatIs.GetTimestamp(),
+			IsActive:  res.WhatIs.GetIsActive(),
+		},
+	}, nil
 }
 
 // UpdateObject updates an object.
 func (s *HighwayServer) UpdateObject(ctx context.Context, req *ot.MsgUpdateObject) (*ot.MsgUpdateObjectResponse, error) {
-	return nil, ErrMethodUnimplemented
+	// Verify that there are requested changes
+	if req.GetAddedFields() == nil && req.GetRemovedFields() == nil {
+		return nil, errors.New("object to register must have fields")
+	}
+
+	// Translate ot fields to otv1
+	addedFields := s.bufToBlockchain(req.GetAddedFields())
+	removedFields := s.bufToBlockchain(req.GetRemovedFields())
+
+	// Build Transaction
+	tx := &otv1.MsgUpdateObject{
+		Creator:       req.GetCreator(),
+		Label:         req.GetLabel(),
+		AddedFields:   addedFields,
+		RemovedFields: removedFields,
+	}
+
+	// Broadcast the message
+	res, err := s.cosmos.BroadcastUpdateObject(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	// decode otv1 fields to ot
+	outputFields := s.decodeObjectDocFields(res.WhatIs.ObjectDoc.Fields)
```

Comment:
should we first check for non nill properties on result to prevent panics

Reply:
Yes, cause len does not short circuit on nil pretty sure